enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,320 @@
+- The ingress capability should be part of the `vCurrent` and `v4.14` baseline sets.
+- The openshift api vendored dependency should be bumped into `cluster-version-operator` repository.
+  - Additionally some more changes will need to be done in the cluster version operator to address [the standalone use case](#standalone-openshift).
+- The openshift api vendored dependency should be bumped in `installer` repository.
+  - This will allow the users of the standalone OpenShift to see the ingress capability. While the enabling of the ingress capability is not supposed to cause any issue, the disabling may have some consequences therefore will be not possible, see [Standalone OpenShift chapter](#standalone-openshift) for more details.
+
+Similar to any other capability, the ingress capability:
+- can be enabled not only as part of the capability sets (`vCurrent`, `v4,14`) but explicitly by using [the additional enabled capability field](https://github.com/openshift/api/blob/3778e7a4a55241c36e66e43cb5d124f44938c094/config/v1/types_cluster_version.go#L433-L438).
+- if enabled, should appear in [the list of the enabled capabilities](https://github.com/openshift/api/blob/3778e7a4a55241c36e66e43cb5d124f44938c094/config/v1/types_cluster_version.go#L445-L448) of `ClusterVersion` object.
+- if enabled, cannot be disabled post install.
+
+#### HyperShift
+
+HyperShift doesn't use the OpenShift installer to spawn the user clusters. HyperShift deploys the cluster version operator to manage the cluster operators though.
+However not all the cluster operators are of use for HyperShift.
+The functionality of some cluster operators is fully replaced by the logic in the hosted control plane (e.g. authentication operator).
+The ingress operator is an example of a cluster operator whose life cycle is shared between the cluster version operator and HyperShift.
+HyperShift doesn't pass all the ingress operator's payload to the cluster version operator.
+Certain manifests are filtered out, most importantly: [the ingress operator's deployment](https://github.com/openshift/hypershift/blob/0a5ec529ce7ae6f238ed27be6eb5757cf70c41f9/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L75).
+The reason for this is that the ingress operator is considered as a part of the hosted control plane hence deployed on the control plane cluster.
+However [the ingress operator's APIs](#ingress-operator-apis) and some RBACs to manage those APIs are still kept in the cluster version operator's payload.
+The ingress capability comes handy here as HyperShift can leverage it to control the remaining (not filtered) payload.
+
+The HyperShift's API which will handle the hosted cluster capabilities is out of the scope of this enhancement. However this enhancement is built with the assumption that the `ClusterVersion`'s capability API (spec and status) is sufficient for any capability implementation in HyperShift.
```

Comment:
```suggestion
HyperShift API to specify capabilities for a hosted cluster is out of the scope of this enhancement. However, this enhancement is built with the assumption that the `ClusterVersion`'s capability API (spec and status) is sufficient for any capability implementation in HyperShift.
```
I'm not sure what "sufficient" means here; do you mean that HyperShift will have some API for specifying capabilities, and the capabilities specified in that API will map directly to the existing capabilities spec and status in the `ClusterVersion` CRD?

Reply:
sufficient in that hypershift can use the capability api to disable the capability in question